Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs

```diff
@@ -0,0 +1,122 @@
+﻿using Blizzard.IO.Core;
+using Blizzard.IO.RabbitMQ.Entities;
+using Blizzard.IO.RabbitMQ.Extensions;
+using Blizzard.IO.Serialization.Json;
+using EasyNetQ;
+using Microsoft.Extensions.Logging;
+
+namespace Blizzard.IO.RabbitMQ.Builders
+{
+    public class NetqRabbitPublisherBuilder<TData> : BaseNetqRabbitBuilder
+    {
+        private RabbitExchange _destinationExchange = null;
+        private ISerializer<TData> _serializer = new JsonSerializer<TData>();
+        private string _routingKey = "/";
+        private bool _isAbstract;
+        private IConverter<RabbitMessageProperties, MessageProperties> _converter;
+        private ILoggerFactory _loggerFactory;
+
+        public NetqRabbitPublisherBuilder(ILoggerFactory loggerFactory) : base(loggerFactory)
+        {
+        }
+
+        public NetqRabbitPublisherBuilder<TData> AddRabbitConnectionProperties(
+            ushort timeout = 10,
+            string product = null,
+            string platform = null,
+            string virtualHost = null,
+            int requestHeartbeat = 10,
+            int prefetchCount = 50,
+            bool publisherConfirms = false,
+            bool persistentMessages = true)
+        {
+            Timeout = timeout;
+            PrefetchCount = prefetchCount;
+            Product = product;
+            Platform = platform;
+            PublisherConfirms = publisherConfirms;
+            RequestHeartbeat = requestHeartbeat;
+            VirtualHost = virtualHost;
+            PersistentMessages = persistentMessages;
+            return this;
+        }
+
+        public NetqRabbitPublisherBuilder<TData> AddHost(string host)
```

Comment:
Why do you have a separate host function?

Reply:
f